### PR TITLE
Harlequin Ichthyosis

### LINKED ISSUE #13497: The HAS_SWEAT_GLANDS Anatomy Flag Doesn't Affect Sweating
https://github.com/d3athrow/vgstation13/blob/Bleeding-Edge/code/modules/mob/thermoregulation.dm#L39

The only code in the repo that references the flag, in `/mob/living/sweat()`, is commented out.

### PR DESCRIPTION
**Pull request by Shadowmech88**
Adds new stage 4 virus symptom, harlequin ichthyosis.
This symptom causes the sufferer's skin to become incredibly thick and dry, with the severity getting worse the longer the sufferer goes without quenching their skin with water.
As their skin dries out and hardens, it loses its ability to effectively disperse heat, causing the sufferer to overheat more easily. However, the sufferer also takes decreased brute damage, due to their skin thickening and hardening.
As the symptom progresses untreated, the sufferer's heat dispersal grows worse. Once their skin dries out enough, they lose the ability to sweat, which further contributes to the risk of overheating.
As the symptom reaches advanced stages, the sufferer will also experience difficulty breathing and frequent bleeding.

Fixes #13497
Since I needed it to work for this.

:cl:
 * rscadd: Added harlequin ichthyosis as a new stage 4 virus symptom, causes overheating among other negative effects the longer you go without quenching your skin with water.

### CODE REVIEW

File: code/modules/mob/thermoregulation.dm [hunk 1]
```diff
@@ -36,13 +40,12 @@
 /mob/living/proc/sweat(var/amount,var/forcesweat = 0)
 	if((status_flags & GODMODE) || (flags & INVULNERABLE))
 		return 1
-	/* buggy as fug
+
 	if(istype(src,/mob/living/carbon/human))
 		var/mob/living/carbon/human/H = src
-		if(!H.species.anatomy_flags & HAS_SWEAT_GLANDS)
+		if(!(H.species.anatomy_flags & HAS_SWEAT_GLANDS))
```
Comment (PJB3005) [inline code/modules/mob/thermoregulation.dm]: Do ` if(~H.species.anatomy_flags & HAS_SWEAT_GLANDS)` instead.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 2]
```diff
@@ -368,6 +368,13 @@
 	//Put out fire
 	if(method == TOUCH)
 		M.ExtinguishMob()
+		if(istype(M, /mob/living/carbon))
```
Comment (PJB3005) [inline code/modules/reagents/Chemistry-Reagents.dm]: `iscarbon`

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/modules/virus2/effect/effect.dm [hunk 3]
```diff
@@ -1309,7 +1309,10 @@ datum/disease2/effect/lubefoot/deactivate(var/mob/living/carbon/mob)
 						if(81 to 100)
 							switch(multiplier)
 								if(81 to 90)
-									to_chat(mob, "<span class='warning'>The thickness of the plate on your chest is making it difficult to breathe.</span>")
+									if(ishuman(mob))
+										var/mob/living/carbon/human/H = mob
+										if(!(H.flags & NO_BREATHE))
```
Comment (PJB3005) [inline code/modules/virus2/effect/effect.dm]: `!(x & y)` -> `~x & y`

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Comment by Probe1** [comment]
All that is needed is to apply water anywhere?

**Comment by Shadowmech88** [comment]
In the same way you'd have to if you were on fire, yes.
Each application of water only reduces the multiplier by 1-3 though, and it gets increased by 10 every time it activates, so you have to be very diligent about keeping it under control.

**Comment by carshalash** [comment]
Harlequin people are gross and look like krispy kreme donuts.

**Comment by kol1th** [comment]
Make their eyes pop out too. :^)

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**